artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -565,11 +565,12 @@ public boolean checkPageFileExists(final int pageNumber) {
       SequentialFileFactory factory = null;
       try {
          factory = checkFileFactory();
+         SequentialFile file = factory.createSequentialFile(fileName);
+         return file.exists();
       } catch (Exception ignored) {
+         assert false : "PagingStoreFactory::newFileFactory never-throws assumption failed.";
```

Comment:
I'm not sure about this `assert` here. First, it references "newFileFactory" which I think is incorrect. Second, I think just returning `false` would be sufficient.

Reply:
See the ticket, I received link to the ticket rather than to this change set so I reacted in the ticket.

Reply:
I agree with @jbertram having assert and rasing errors is not nice. Its much cleaner to return boolean with true false if it was successful

Reply:
the assert gives flexibility:
   in prod environments it can erased (like it would not exist in the sources)
   in dev environments it stops so one have proof out of contact situation actually happened

I went for assert as the contract is not evident. Returning mere false means that contract it clear and understood.  

If anybody helps to clarify contract then removing assert is super simple change to accept. 

  Hope it helps

Reply:
To be clear i am -1 this as assert raises a runtime exception, you may as well just leave it with npe if youre doing that.  Other options have been chaired by myself and @jbertram and im sure there is alternatives too